src/components/WeatherShowcase/WeatherShowcase.tsx

```diff
@@ -1,10 +1,11 @@
+/* eslint-disable no-console */
 import React, { useEffect, useState } from "react";
 import axios from "axios";
 
-type WeatherData = {
+interface WeatherData {
```

Comment:
It's interesting you changed this, as it was actually correct as it was. 

It's not official TypeScript best practice, but in the projects I've worked on, we have always made a simple distinction between when to use `interface` and `type`. 

We use `interface` for function arguments and component props. Because it's the way we interact or interface with that function.  

The convention is

```
interface FormatStringWithOptionsArgs  { ...

const formatStringWithOptions = ({ string, options }: FormatStringWithOptionsArgs) => {...

interface ButtonProps {...

function Button({ onClick, type, ... }: ButtonProps) { ...
```

types are used everywhere else for data being passed around the application.